classes/class.ilObjOpenCast.php

```diff
@@ -30,6 +31,8 @@ public function __construct(int $a_ref_id = 0)
         global $DIC;
         $this->ctrl = $DIC->ctrl();
         parent::__construct($a_ref_id);
+        $this->cache = Init::init($DIC)->get(Services::class);
```

Comment:
Do we need to instantiate `Init` twice?  
Instead, we could reuse a single instance, or am I not seeing something here?
```suggestion
        $init_instance = Init::init($DIC);
        $this->cache = $init_instance->get(Services::class);
        $this->series_repository = $init_instance->get(SeriesAPIRepository::class);
```

Reply:
@chfsx, how about here?